modules/core/src/main/scala/com.snowplowanalytics.snowplow.lakes/processing/LakeWriter.scala

```diff
@@ -164,25 +155,21 @@ object LakeWriter {
 
     def commit(viewName: String): F[Unit] =
       for {
-        df <- mutexForUnioning.lock.surround {
-                SparkUtils.prepareFinalDataFrame(spark, viewName, writerParallelism)
-              }
-        _ <- mutexForWriting.lock
+        df <- SparkUtils.prepareFinalDataFrame(spark, viewName, writerParallelism)
```

Comment:
Is it okay if `prepareFinalDataFrame` is run by two different windows from now on ?

Reply:
Yes.  After these changes, `prepareFinalDataFrame` does not actually do anything.  In Spark, DataFrames are lazy so they are not evaluated until needed.  Before these changes, we force-evaluated the final DataFrame by calling `.localCheckpoint()`.

For the same reason, you might have noticed that `prepareFinalDataFrame` no longer uses `Sync[F].blocking{ }`.